.github/workflows/release.yml

```diff
@@ -6,28 +6,113 @@ on:
         with:
           go-version: 1.19
           check-latest: true
 
-      - uses: sigstore/cosign-installer@9becc617647dfa20ae7b1151972e9b3a2c338a2b # v2.8.1
-      - uses: anchore/sbom-action/download-syft@06e109483e6aa305a2b2395eabae554e51530e1d # v0.13.1
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@9becc617647dfa20ae7b1151972e9b3a2c338a2b # v2.8.1
 
-      - uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+      - name: Install Syft
+        uses: anchore/sbom-action/download-syft@06e109483e6aa305a2b2395eabae554e51530e1d # v0.13.1
 
-      - uses: goreleaser/goreleaser-action@8f67e590f2d095516493f017008adc464e63adb1 # v4.1.0
-        with:
-          install-only: true
+      - name: Checkout
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+
+      # - name: Set LDFLAGS
+      #   id: ldflags
+      #   run: |
+      #     echo "GO_FLAGS=( echo "-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=${{needs.args.outputs.version}} \
+      #     -X sigs.k8s.io/release-utils/version.gitCommit==${{needs.args.outputs.commit}} \
+      #     -X sigs.k8s.io/release-utils/version.gitTreeState==${{needs.args.outputs.tree-state}} \
+      #     -X sigs.k8s.io/release-utils/version.buildDate==${{needs.args.outputs.commit-date}}" )" >> "$GITHUB_ENV"
 
-      - name: release
-        run: make release
+      - name: Run GoReleaser
+        id: run-goreleaser
+        uses: goreleaser/goreleaser-action@8f67e590f2d095516493f017008adc464e63adb1 # v4.1.0
+        with:
+          version: latest
+          args: release --rm-dist
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          LDFLAGS: echo "-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=${{needs.args.outputs.version}} \
+            -X sigs.k8s.io/release-utils/version.gitCommit==${{needs.args.outputs.commit}} \
+            -X sigs.k8s.io/release-utils/version.gitTreeState==${{needs.args.outputs.tree-state}} \
+            -X sigs.k8s.io/release-utils/version.buildDate==${{needs.args.outputs.commit-date}}" #${{ env.GO_FLAGS }}
+
+      - name: Generate subject
+        id: hash
+        env:
+          ARTIFACTS: "${{ steps.run-goreleaser.outputs.artifacts }}"
+        run: |
+          set -euo pipefail
+          checksum_file=$(echo "$ARTIFACTS" | jq -r '.[] | select (.type=="Checksum") | .path')
+          echo "hashes=$(cat $checksum_file | base64 -w0)" >> "$GITHUB_OUTPUT"
+
+  provenance:
+    needs: release
+    permissions:
+      actions: read # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.4.0
+    with:
+      base64-subjects: "${{ needs.release.outputs.hashes }}"
+      upload-assets: false # do not upload to a new release
```

Comment:
any reason to set this to `false`? If set to `true`, you should not need the `release-provenance` job.
/cc @ianlewis

Reply:
I don't think we support the `draft: true` option that is used below. It seems like they are trying to preserve the previous behavior of running `goreleaser release`.

https://github.com/sigstore/helm-sigstore/blob/60bfeab071df7ac8d7405f8d98d3ba2b00488853/.goreleaser.yml#L80

That may be the reason it's being done this way?

I wonder if `goreleaser release` or prior call to `softprops/action-gh-release` creates a draft release, will running `softprops/action-gh-release` afterward to upload the provenance without setting `draft: true` fully publish the release (i.e. unflip the draft bit)?

Reply:
That's strange if it created two releases for the same tag. In any case it seems they may want to create the release as a draft and manually publish it.